src/updater.ts

```diff
@@ -92,81 +91,66 @@ export default class Updater {
     const startTime = Date.now();
 
     // Get game notifications
-    const gameNotifications = await mapAsync(Game.getGames(), async (game, index, games) => {
-      const updates = await this.updateGame(game);
+    await Game.getGames().reduce(async (prevUpdatesHandle: Promise<void>, game: Game, index) => {
+      // Wait for the previous game update to finish
+      await prevUpdatesHandle;
 
-      // If there are more games in this update cycle, delay them by the specified amount
-      if (index < games.length - 1) {
+      // If this is not the first game, delay the update
+      if (index !== 0) {
         await sleep(this.gameIntervalMs);
       }
 
-      return updates;
-    });
-
-    // Combine the game notifications
-    let notifications: Notification[] = mergeArrays(gameNotifications);
-
-    if (notifications.length > 0) {
-      // Sort the notifications by their date, from old to new.
-      notifications = sort(notifications);
-
-      const endPollTime = Date.now();
-      const pollTime = endPollTime - startTime;
-      this.logger.info(
-        `Found ${notifications.length} posts in ${pollTime} ms. Notifying channels...`,
-      );
-
-      // Update time
-      this.saveDate(notifications[notifications.length - 1].timestamp);
+      // Get the updates for the current game
+      await this.updateGame(game);
+    }, Promise.resolve());
```

Comment:
I don't find this more readable than a simple for loop

Reply:
Fixed